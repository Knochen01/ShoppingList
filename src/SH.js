import React, { useState } from "react";
import Item2 from "./Item2";
import NewItem2 from "./NewItem2";
import uuid from 'react-uuid';


const SH = () => {

    const INITIAL_STATE = [
        {id: uuid(), name:"PlaceHolder", qty:3}
    ]

    const [items, setItems] = useState(INITIAL_STATE )

    const addItem = (name, qty) => {
        setItems(items => [...items, { name, qty,  id: uuid() }] )
    }

    return (
        <div>
            <h3>ShoppingList</h3>
            <NewItem2 addItem={addItem} />
            {items.map(({ name, qty }) => <Item2 name={name} qty={qty} id={uuid()} key={uuid()} />)}
        </div>
    )
}
export default SH;