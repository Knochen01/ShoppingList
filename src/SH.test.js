import React from "react";
import SH from "./SH";
import { render, fireEvent} from '@testing-library/react';

// Render
test("it Should Render on the Screen", () => {
    render(<SH />)
})
// SnapShot
it("should Match the Snapshot", () => {
    const { asFragment } = render(<SH />)
    expect(asFragment()).toMatchSnapshot()
})
// New Item
test("should add New Item", () => {
    const { queryByText, getByLabelText } = render(<SH />)
    const input = getByLabelText("Add Item:")
    const btn = queryByText("Add to List")
    expect(queryByText("ProductName: Chocolate")).not.toBeInTheDocument()
    fireEvent.change(input, { target: { value: "Chocolate"}} )
    fireEvent.click(btn)
    expect(queryByText("ProductName: Chocolate")).toBeInTheDocument()
})