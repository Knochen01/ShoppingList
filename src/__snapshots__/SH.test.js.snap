// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Match the Snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      ShoppingList
    </h3>
    <div>
      <form>
        <label
          for="addItem"
        >
          Add Item:
        </label>
        <input
          id="addItem"
          name="item"
          placeholder="Add Item"
          type="text"
          value=""
        />
        <label>
          Add Quantity:
        </label>
        <input
          id="Quantity"
          min="0"
          name="qty"
          placeholder="Quantity"
          type="number"
          value=""
        />
        <button>
          Add to List
        </button>
      </form>
    </div>
    <div>
      <ul
        style="list-style: none;"
      >
        <li>
          ProductName: PlaceHolder
        </li>
        <li>
          Quantity: 3
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
